codice/ddf
# DDF-2189: Updated code to compare 2 metacards using attributes instead of using the hashcodes

#### What does this PR do?
Changed `MetacardComparator` to fix a few issues with the comparison which were causing
NullPointerException when a resource was retrieved from the cache after the initial download.

Also added `equals()` and `hashCode()` implementations to `AttributeImpl` and re-enabled
some integration tests.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@shaundmorris 
@bdeining 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@pklinef
#### How should this be tested?
* Run the full suite of integration tests.
* Try download the same metacard resource twice and make sure no NPE is thrown and the resource downloads successfully.

#### Any background context you want to provide?
#### What are the relevant tickets?
#### Screenshots (if appropriate)
#### Checklist:
